### PR TITLE
mempool: Fix the benchmarks

### PR DESCRIPTION
**Pull request by otrack**
The mempool [benchmarks](https://github.com/cometbft/cometbft/blob/main/mempool/bench_test.go) were invalid due to an incorrect transaction payload.  To see this, one may run `go test -v -count 1 -bench ^BenchmarkCheckTx$ -test.run ^$` against `main` from the `mempool` directory. These modificationss fix such a problem and add a new benchmark to exercice`mempool:update`.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: mempool/bench_test.go [hunk 1]
```diff
@@ -37,14 +43,12 @@ func BenchmarkCheckTx(b *testing.B) {
 	mp, cleanup := newMempoolWithApp(cc)
 	defer cleanup()
 
-	mp.config.Size = 1000000
+	mp.config.Size = 100_000_000
```
Comment (sergio-mena) [inline mempool/bench_test.go:46]: The # of zeros changed here, is it intended?

**Review by sergio-mena — APPROVED** [review]
LGTM. Thanks for this

**Review by sergio-mena — APPROVED** [review]
(you might want to update to `main` once #984 is in)

**Comment by hvanz** [comment]
@Mergifyio backport v0.38.x

**Pull request merged**